hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -28,9 +28,18 @@ def _compute_sheet(self):
             if timesheet.sheet_id != sheet:
                 timesheet.sheet_id = sheet
 
+    def _check_sheet_company_id(self, sheet_id):
```

Comment:
Why not use the `api.contrains` and thus avoids to add method in create?

Reply:
This check, if written as a constraint, is not triggered during the creation. The reason is that `company_id` in the analytic line is a readonly field.

At the end I need the check only during the creation only.
I opened another PR: https://github.com/OCA/hr-timesheet/pull/220, that PR adds the constraint you are mentioning. You may want to look at that first. In a certain point of view, this PR covers a case that is not covered by https://github.com/OCA/hr-timesheet/pull/220.

Reply:
Ah, good to know, there a lot of PRs haha